### PR TITLE
[Enhancement] Add be_threads info into infomation schema

### LINKED ISSUE #20445: [Enhancement] Expose BE threads information to information_schema table
## Enhancement

```
mysql> select * from be_threads;
+-------+-------------+-----------------------------+-----------------+---------+------+----------------+
| BE_ID | GROUP       | NAME                        | PTHREAD_ID      | OS_TID  | IDLE | FINISHED_TASKS |
+-------+-------------+-----------------------------+-----------------+---------+------+----------------+
| 10004 | pipeline    | pipeline_poller             | 140547443349248 | 2116137 |    1 |              0 |
| 10004 | pipeline    | pipeline_poller             | 140547661559552 | 2116111 |    1 |              0 |
| 10004 | thread pool | compact_pool                | 140542684686080 | 2116704 |    1 |              0 |
| 10004 | thread pool | up_candidates               | 140542701471488 | 2116702 |    1 |             16 |
| 10004 | thread pool | clone                       | 140543154677504 | 2116648 |    1 |              0 |
| 10004 | thread pool | clone                       | 140543163070208 | 2116647 |    1 |              0 |
| 10004 | thread pool | drop_auto_increment_map_dir | 140543171462912 | 2116646 |    1 |              0 |
| 10004 | thread pool | update_tablet_meta_info     | 140543179855616 | 2116645 |    1 |              0 |
| 10004 | thread pool | move_dir                    | 140543188248320 | 2116644 |    1 |              0 |
| 10004 | thread pool | release_snapshot            | 140543196641024 | 2116643 |    1 |              0 |
| 10004 | thread pool | release_snapshot            | 140543205033728 | 2116642 |    1 |              0 |
| 10004 | thread pool | release_snapshot            | 140543213426432 | 2116641 |    1 |              0 |
...

mysql> select be_id, name, idle, count(*), sum(finished_tasks) from be_threads group by be_id, name, idle order by name;
+-------+-----------------------------+------+----------+---------------------+
| be_id | name                        | idle | count(*) | sum(finished_tasks) |
+-------+-----------------------------+------+----------+---------------------+
| 10004 | alter_tablet                |    1 |        3 |                   0 |
| 10004 | check_consistency           |    1 |        1 |                   0 |
| 10004 | clear_transaction           |    1 |        1 |                   0 |
| 10004 | clone                       |    1 |        2 |                   0 |
| 10004 | compact_pool                |    1 |        1 |                   0 |
| 10004 | con_scan_io                 |    1 |      192 |                   0 |
| 10004 | con_wg_scan_io              |    1 |      192 |                   0 |
| 10004 | create_tablet               |    1 |        3 |                   0 |
| 10004 | delta_writer                |    1 |        8 |                   0 |
| 10004 | download                    |    1 |        1 |                   0 |
| 10004 | drop                        |    1 |        3 |                   0 |
| 10004 | drop_auto_increment_map_dir |    1 |        1 |                   0 |
| 10004 | ex_state_report             |    1 |        2 |                   7 |
| 10004 | fragment_mgr                |    1 |       64 |                   0 |
| 10004 | make_snapshot               |    1 |        5 |                   0 |
| 10004 | memtable_flush              |    1 |        2 |                   0 |
| 10004 | meta-flush                  |    1 |        1 |                   0 |
| 10004 | move_dir                    |    1 |        1 |                   0 |
| 10004 | pip_executor                |    1 |       24 |                   0 |
| 10004 | pip_scan_io                 |    1 |       24 |                   0 |
| 10004 | pip_wg_executor             |    1 |       23 |                  69 |
| 10004 | pip_wg_executor             |    0 |        1 |                   3 |
| 10004 | pip_wg_scan_io              |    1 |       24 |                   5 |
| 10004 | pipeline_poller             |    1 |        2 |                   0 |
| 10004 | publish_version             |    1 |        8 |                   0 |
| 10004 | release_snapshot            |    1 |        5 |                   0 |
| 10004 | segment_flush               |    1 |        2 |                   0 |
| 10004 | segment_replicate           |    1 |        2 |                   0 |
| 10004 | storage_medium_migrate      |    1 |        1 |                   0 |
| 10004 | up_candidates               |    1 |        1 |                  16 |
| 10004 | update_tablet_meta_info     |    1 |        1 |                   0 |
| 10004 | upload                      |    1 |        1 |                   0 |
+-------+-----------------------------+------+----------+---------------------+
32 rows in set (0.07 sec)

```

### PR DESCRIPTION
**Pull request by decster**
## What type of PR is this：
- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #20445

## Problem Summary(Required) ：

This PR adds backend's thread information into information schema, currently only threads created in ThreadPool are listed, more threads will be added later.

Example usage:
```
mysql> select * from be_threads;
+-------+-------------+-----------------------------+-----------------+---------+------+----------------+
| BE_ID | GROUP       | NAME                        | PTHREAD_ID      |    TID  | IDLE | FINISHED_TASKS |
+-------+-------------+-----------------------------+-----------------+---------+------+----------------+
| 10004 | pipeline    | pipeline_poller             | 140547443349248 | 2116137 |    1 |              0 |
| 10004 | pipeline    | pipeline_poller             | 140547661559552 | 2116111 |    1 |              0 |
| 10004 | thread pool | compact_pool                | 140542684686080 | 2116704 |    1 |              0 |
| 10004 | thread pool | up_candidates               | 140542701471488 | 2116702 |    1 |             16 |
| 10004 | thread pool | clone                       | 140543154677504 | 2116648 |    1 |              0 |
| 10004 | thread pool | clone                       | 140543163070208 | 2116647 |    1 |              0 |
| 10004 | thread pool | drop_auto_increment_map_dir | 140543171462912 | 2116646 |    1 |              0 |
| 10004 | thread pool | update_tablet_meta_info     | 140543179855616 | 2116645 |    1 |              0 |
| 10004 | thread pool | move_dir                    | 140543188248320 | 2116644 |    1 |              0 |
| 10004 | thread pool | release_snapshot            | 140543196641024 | 2116643 |    1 |              0 |
| 10004 | thread pool | release_snapshot            | 140543205033728 | 2116642 |    1 |              0 |
| 10004 | thread pool | release_snapshot            | 140543213426432 | 2116641 |    1 |              0 |
...

mysql> select be_id, name, idle, count(*), sum(finished_tasks) from be_threads group by be_id, name, idle order by name;
+-------+-----------------------------+------+----------+---------------------+
| be_id | name                        | idle | count(*) | sum(finished_tasks) |
+-------+-----------------------------+------+----------+---------------------+
| 10004 | alter_tablet                |    1 |        3 |                   0 |
| 10004 | check_consistency           |    1 |        1 |                   0 |
| 10004 | clear_transaction           |    1 |        1 |                   0 |
| 10004 | clone                       |    1 |        2 |                   0 |
| 10004 | compact_pool                |    1 |        1 |                   0 |
| 10004 | con_scan_io                 |    1 |      192 |                   0 |
| 10004 | con_wg_scan_io              |    1 |      192 |                   0 |
| 10004 | create_tablet               |    1 |        3 |                   0 |
| 10004 | delta_writer                |    1 |        8 |                   0 |
| 10004 | download                    |    1 |        1 |                   0 |
| 10004 | drop                        |    1 |        3 |                   0 |
| 10004 | drop_auto_increment_map_dir |    1 |        1 |                   0 |
| 10004 | ex_state_report             |    1 |        2 |                   7 |
| 10004 | fragment_mgr                |    1 |       64 |                   0 |
| 10004 | make_snapshot               |    1 |        5 |                   0 |
| 10004 | memtable_flush              |    1 |        2 |                   0 |
| 10004 | meta-flush                  |    1 |        1 |                   0 |
| 10004 | move_dir                    |    1 |        1 |                   0 |
| 10004 | pip_executor                |    1 |       24 |                   0 |
| 10004 | pip_scan_io                 |    1 |       24 |                   0 |
| 10004 | pip_wg_executor             |    1 |       23 |                  69 |
| 10004 | pip_wg_executor             |    0 |        1 |                   3 |
| 10004 | pip_wg_scan_io              |    1 |       24 |                   5 |
| 10004 | pipeline_poller             |    1 |        2 |                   0 |
| 10004 | publish_version             |    1 |        8 |                   0 |
| 10004 | release_snapshot            |    1 |        5 |                   0 |
| 10004 | segment_flush               |    1 |        2 |                   0 |
| 10004 | segment_replicate           |    1 |        2 |                   0 |
| 10004 | storage_medium_migrate      |    1 |        1 |                   0 |
| 10004 | up_candidates               |    1 |        1 |                  16 |
| 10004 | update_tablet_meta_info     |    1 |        1 |                   0 |
| 10004 | upload                      |    1 |        1 |                   0 |
+-------+-----------------------------+------+----------+---------------------+
32 rows in set (0.07 sec)
```

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto backported to target branch
  - [x] 3.0
  - [x] 2.5
  - [x] 2.4
  - [ ] 2.3

### CODE REVIEW

File: be/src/exec/schema_scanner/schema_be_threads_scanner.cpp [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+#include "runtime/string_value.h"
+#include "types/logical_type.h"
+#include "util/thread.h"
+
+namespace starrocks {
+
+SchemaScanner::ColumnDesc SchemaBeThreadsScanner::_s_columns[] = {
+        {"BE_ID", TYPE_BIGINT, sizeof(int64_t), false},          {"GROUP", TYPE_VARCHAR, sizeof(StringValue), false},
+        {"NAME", TYPE_VARCHAR, sizeof(StringValue), false},      {"PTHREAD_ID", TYPE_BIGINT, sizeof(int64_t), false},
+        {"OS_TID", TYPE_BIGINT, sizeof(int64_t), false},         {"IDLE", TYPE_BOOLEAN, 1, false},
```
Comment (chaoyli) [inline be/src/exec/schema_scanner/schema_be_threads_scanner.cpp:29]: TID is enough.

Comment (decster) [reply be/src/exec/schema_scanner/schema_be_threads_scanner.cpp]: changed

**Review by chaoyli — APPROVED** [review]

**Review by Astralidea — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.0

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-2.5

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-2.4

**Pull request merged**